### PR TITLE
Update documentation of the messages module [ECR-3275]

### PR DESCRIPTION
**Pull request by alekseysidorov**

### CODE REVIEW

File: exonum/src/messages/types.rs [hunk 1]
```diff
@@ -27,7 +27,9 @@ use crate::{
     proto::schema::consensus,
 };
 
-/// Container for the signed messages.
+/// Protobuf based container for the any signed messages.
```
Comment (ozkriff) [inline exonum/src/messages/types.rs]: "the" here looks a little bit suspicious to me

File: exonum/src/messages/mod.rs [hunk 2]
```diff
@@ -29,7 +29,63 @@
...
+//! };
+//!
+//! # fn send<T>(_: T) {}
+//! let keypair = crypto::gen_keypair();
+//! // Let we have a some Status message.
+//! let payload = Status {
+//!     height: Height(15),
+//!     last_hash: Hash::zero(),
+//! };
+//! // With a some keypair we can sign it and get trusted status message.
```
Comment (ozkriff) [inline exonum/src/messages/mod.rs]: unneeded "a"

File: exonum/src/messages/mod.rs [hunk 3]
```diff
@@ -29,7 +29,63 @@
...
+//! let signed_payload = Verified::from_value(payload, keypair.0, &keypair.1);
+//! // Later we can convert it to raw signed message and sends by the network.
```
Comment (ozkriff) [inline exonum/src/messages/mod.rs]: "to **a** raw signed message"?

"sends" -> "send"

File: exonum/src/messages/mod.rs [hunk 4]
```diff
@@ -29,7 +29,63 @@
...
+//!
+//! # fn get_signed_message() -> SignedMessage {
+//! #   let keypair = crypto::gen_keypair();
+//! #   let payload = Status {
+//! #       height: Height(15),
+//! #       last_hash: Hash::zero(),
+//! #   };
+//! #   Verified::from_value(payload, keypair.0, &keypair.1).into_raw()
+//! # }
+//! // For example we have got a some signed message.
```
Comment (ozkriff) [inline exonum/src/messages/mod.rs]: unneeded "a" (or "some")

File: exonum/src/messages/mod.rs (hunk 2 above)
Comment (ozkriff) [inline exonum/src/messages/mod.rs]:
```suggestion
//! // With some keypair we can sign it and get trusted status message.
```

File: exonum/src/messages/mod.rs (hunk 3 above)
Comment (ozkriff) [inline exonum/src/messages/mod.rs]:
```suggestion
//! // Later we can convert it to a raw signed message and send it by the network.
```

File: exonum/src/messages/mod.rs (hunk 4 above)
Comment (ozkriff) [inline exonum/src/messages/mod.rs]:
```suggestion
//! // For example, we have got some signed message.
```

File: exonum/src/messages/mod.rs [hunk 5]
```diff
@@ -29,7 +29,63 @@
...
+//! ```
+//! use exonum::{
+//!     crypto::{self, Hash},
+//!     helpers::Height,
+//!     messages::{Status, Verified},
+//! };
+//!
+//! # fn send<T>(_: T) {}
+//! let keypair = crypto::gen_keypair();
+//! // Let we have a some Status message.
```
Comment (ozkriff) [inline exonum/src/messages/mod.rs]:
```suggestion
//! // We have some Status message.
```

File: exonum/src/messages/mod.rs [hunk 6]
```diff
@@ -29,7 +29,63 @@
...
+//!     last_hash: Hash::zero(),
+//! };
+//! // With a some keypair we can sign it and get trusted status message.
+//! let signed_payload = Verified::from_value(payload, keypair.0, &keypair.1);
+//! // Later we can convert it to raw signed message and sends by the network.
+//! let raw_signed_message = signed_payload.into_raw();
+//! send(raw_signed_message);
 //! ```
+//!
+//! The procedure of a signed message verification.
```
Comment (Maria-Nosyk) [inline exonum/src/messages/mod.rs]: The procedure of verification of a signed message is as follows:

File: exonum/src/messages/mod.rs (hunk 4 above)
Comment (Maria-Nosyk) [inline exonum/src/messages/mod.rs]: Assume you have some signed message.

File: exonum/src/messages/mod.rs [hunk 7]
```diff
@@ -29,7 +29,63 @@
...
+//! #   let payload = Status {
+//! #       height: Height(15),
+//! #       last_hash: Hash::zero(),
+//! #   };
+//! #   Verified::from_value(payload, keypair.0, &keypair.1).into_raw()
+//! # }
+//! // For example we have got a some signed message.
+//! let raw: SignedMessage = get_signed_message();
+//! // We know that this is one type of `ExonumMessage`, so we can try to
+//! // verify its signature and try to convert it into `ExonumMessage`.
```
Comment (Maria-Nosyk) [inline exonum/src/messages/mod.rs]: //! // You know that this is a type of `ExonumMessage`, so you can 
//! // verify its signature and convert it into `ExonumMessage`.

File: exonum/src/messages/mod.rs [hunk 8]
```diff
@@ -29,7 +29,63 @@
...
+//! let verified = raw.into_verified::<ExonumMessage>().expect("verification failed");
+//! // And further to look whether it is for example the status message.
```
Comment (Maria-Nosyk) [inline exonum/src/messages/mod.rs]: Further, check whether it is a "Status" message.

File: exonum/src/messages/types.rs (hunk 1 above)
Comment (Maria-Nosyk) [inline exonum/src/messages/types.rs]: for any signed message.

File: exonum/src/messages/mod.rs [hunk 9]
```diff
@@ -194,7 +251,7 @@ impl Message {
     }
 
     /// Checks buffer and return instance of `Message`.
```
Comment (Maria-Nosyk) [inline exonum/src/messages/mod.rs]: Checks... and returns...
or
Check... and return...

Comment (alekseysidorov) [reply exonum/src/messages/mod.rs]: Fixed

**Pull request merged**